components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1259,9 +1259,16 @@ where
                 let raft_msg = self.fsm.peer.build_raft_messages(self.ctx, vec![msg]);
                 self.fsm.peer.send_raft_messages(self.ctx, raft_msg);
             }
-            CasualMessage::SnapshotApplied => {
+            CasualMessage::SnapshotApplied { peer_id, tombstone } => {
                 self.fsm.has_ready = true;
+                // If failed on applying snapshot, it should tomebstone the peer.
+                self.fsm.peer.should_tombstone =
```

Comment:
To prevent the loss of messages in case of sending failures due to unexpected issues.
The flag `should_tombstone` is used to ensure that the abnormal peer continues to send the TombStone message to the leader until it is removed from the raft group.

Reply:
still can send it here

Reply:
FYI, the abnormal peer will stop sending messages once it is destroyed, which is triggered by receiving a `RaftMessage` with the flag `RaftMessage::is_tombstone` set to true.